# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/systemservices/objc_video_capture/device_info.mm
# Commit: 703360f0b861
# Full Hash: 703360f0b8611916dfb3724eda7f35345d3f4be7
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2022-12-13 04:11:09
# Regressor Bug: 1451394
# File Overlap Count: 1
# Description:
#   Bug 1451394 - Integrate with the libwebrtc camera backend for Mac. r=webrtc-reviewers,jib
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D163684
# ==============================================================================

diff -r 49002523b70d -r 703360f0b861 dom/media/systemservices/objc_video_capture/device_info.mm
--- a/dom/media/systemservices/objc_video_capture/device_info.mm	Mon Dec 12 15:47:22 2022 +0000
+++ b/dom/media/systemservices/objc_video_capture/device_info.mm	Mon Dec 12 15:47:22 2022 +0000
@@ -19,8 +19,11 @@
 #include "device_info.h"
 #include "device_info_objc.h"
 #include "modules/video_capture/video_capture_impl.h"
+#include "mozilla/StaticPrefs_media.h"
+#include "objc_video_capture/device_info_avfoundation.h"
 #include "rtc_base/logging.h"
 
+using namespace mozilla;
 using namespace webrtc;
 using namespace videocapturemodule;
 
@@ -32,7 +35,12 @@
   RTC_LOG(LS_ERROR) << __FUNCTION__ << " is not supported on the iOS platform."; \
   return -1;
 
-VideoCaptureModule::DeviceInfo* VideoCaptureImpl::CreateDeviceInfo() { return new DeviceInfoIos(); }
+VideoCaptureModule::DeviceInfo* VideoCaptureImpl::CreateDeviceInfo() {
+  if (StaticPrefs::media_getusermedia_camera_macavf_enabled_AtStartup()) {
+    return new DeviceInfoAvFoundation();
+  }
+  return new DeviceInfoIos();
+}
 
 DeviceInfoIos::DeviceInfoIos() { this->Init(); }
 